ollama_dl.py

```diff
@@ -40,43 +39,82 @@ class DownloadJob:
     blob_url: str
     size: int
 
+async def download_with_resume(client: httpx.AsyncClient, url: str, dest_path: pathlib.Path, total_size: int, progress, task):
+    temp_path = dest_path.with_suffix(f".tmp-{time.time()}")
+    headers = {}
+    mode = "wb"
+    
+    if temp_path.exists():
+        current_size = temp_path.stat().st_size
+        if current_size < total_size:
+            headers["Range"] = f"bytes={current_size}-"
+            mode = "ab"
+            progress.update(task, completed=current_size)
+        else:
+            temp_path.unlink()
+    
+    try:
+        async with client.stream("GET", url, headers=headers, follow_redirects=True) as resp:
+            resp.raise_for_status()
+            log.debug(f"Response headers: {resp.headers}")
+            content_length = int(resp.headers.get('Content-Length', '0'))
+            log.debug(f"Content-Length: {content_length}, Expected total size: {total_size}")
+            
+            with temp_path.open(mode) as f:
+                async for chunk in resp.aiter_bytes(8192):
```

Comment:
8192 bytes is _tiny_ for multi-gigabyte files. 

The large buffer size was chosen on purpose to avoid as many `read` OS calls. See e.g. https://github.com/pytorch/pytorch/pull/116536#discussion_r1440334502 for an explanation.